src/Merchants/Content/Merchant/Subscriber/ConfirmPageLoadedSubscriber.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Shopware\Production\Merchants\Content\Merchant\Subscriber;
+
+use Shopware\Core\Framework\DataAbstractionLayer\EntityCollection;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Shopware\Production\Merchants\Content\Merchant\MerchantEntity;
+use Shopware\Storefront\Page\Checkout\Confirm\CheckoutConfirmPage;
+use Shopware\Storefront\Page\Checkout\Confirm\CheckoutConfirmPageLoadedEvent;
+
+class ConfirmPageLoadedSubscriber
+{
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $productRepository;
+
+    public function __construct(EntityRepositoryInterface $productRepository)
+    {
+        $this->productRepository = $productRepository;
+    }
+
+    public function __invoke(CheckoutConfirmPageLoadedEvent $event)
+    {
+        $merchant = $this->getMerchant($event->getPage(), $event->getSalesChannelContext());
+
+        $this->filterShippingMethods($merchant, $event->getPage()->getShippingMethods(), $event->getSalesChannelContext());
+    }
+
+    private function getMerchant(CheckoutConfirmPage $page, SalesChannelContext $context): MerchantEntity
+    {
+        $productId = $page->getCart()->getLineItems()->first()->getReferencedId();
+
+        $criteria = new Criteria([$productId]);
+        $criteria->addAssociation('merchants.shippingMethods');
+
+        return $this->productRepository->search($criteria, $context->getContext())->first()->getExtension('merchants')->first();
+    }
+
+    private function filterShippingMethods(MerchantEntity $merchant, EntityCollection $entityCollection, SalesChannelContext $context): void
```

Comment:
Might want to rename `$entityCollection` to something like `$shippingMethods` ? Same with `$item`.